spmd/tensor/device_mesh.py

```diff
@@ -211,44 +217,78 @@ def get_rank(self) -> int:
         return get_rank()
 
     def scatter(
-        self, tensors: List[torch.Tensor], src: int = 0
+        self, scatter_list: List[torch.Tensor], mesh_dim: int = 0
     ) -> torch.Tensor:
-        return scatter(tensors, src=src)
+        to_scatter = [tensor.contiguous() for tensor in scatter_list]
```

Comment:
So does this solve the scatter errors observed in rowwise sharding of Linear module?

Reply:
Yep I manually checked it, this should solve it, the `nn.functional.scatter` have two bugs:
1. no contiguous call
2. checking local rank instead of global rank
we gonna need to fix it in the core